app/controllers/SessionsCategoryController.scala

```diff
@@ -0,0 +1,258 @@
+                  BadRequest("Primary category cannot be added due to some error")
+                }
+              }
+          }
+        }
+    }
+  }
+
+  def addSubCategory(categoryName: String, subCategory: String): Action[AnyContent] = adminAction.async { implicit request =>
+    val cleanedSubCategoryName = subCategory.trim
+
+    if (cleanedSubCategoryName.isEmpty) {
+      Future.successful(BadRequest("Subcategory cannot be empty"))
+    } else {
+      categoriesRepository
+        .getCategories
+        .flatMap { categories =>
+          categories.find(_.categoryName == categoryName)
+            .fold {
+              Logger.info(s"No primary category found for category name $categoryName")
+              Future.successful(BadRequest("No primary category found."))
+            } { categoryInfo =>
+              val newSubCategory = categoryInfo.subCategory.find(_.toLowerCase == cleanedSubCategoryName.toLowerCase)
+              newSubCategory.fold {
+                val subCategoryInfo = CategoryInfo(categoryName, List(cleanedSubCategoryName), categoryInfo._id)
+                categoriesRepository
+                  .upsert(subCategoryInfo)
+                  .map { result =>
+                    if (result.ok) {
+                      Logger.info(s"Sub-category was successfully added $subCategory")
+                      Ok("Sub-category was successfully added")
+                    } else {
+                      Logger.error(s"Something went wrong while adding sub-category $subCategory")
+                      BadRequest("Unsuccessful sub-category added")
+                    }
+                  }
+              } { _ =>
+                Future.successful(BadRequest("Error! Sub-category already exists"))
+              }
+            }
+        }
+    }
+  }
+
+  def modifyPrimaryCategory(categoryId: String, newCategoryName: String): Action[AnyContent] = superUserAction.async { implicit request =>
+    if (newCategoryName.trim.isEmpty) {
+      Future.successful(BadRequest("Modify primary category cannot be empty"))
+    } else {
+      categoriesRepository
+        .getCategories
+        .flatMap { categories =>
+          val category = categories.filter(_._id.stringify == categoryId).head
+          sessionsRepository.updateCategoryOnChange(category.categoryName, newCategoryName).flatMap { session =>
+            if (session.ok) {
+              categoriesRepository.modifyPrimaryCategory(category._id.stringify, newCategoryName).map { result =>
+                if (result.ok) {
+                  Logger.info(s"Primary category was successfully modified $newCategoryName")
+                  Ok("Primary category was successfully modified")
+                } else {
+                  Logger.error(s"Something went wrong while modifying primary category $newCategoryName")
+                  BadRequest("Unsuccessful attempt to modify primary category")
```

Comment:
Replace "Unsuccessful attempt to modify primary category" with "Primary category cannot be modified due to some error"